panel/links.py

```diff
@@ -92,23 +92,27 @@ def _process_links(cls, root_view, root_model):
         if not isinstance(root_view, Panel) or not root_model:
             return
 
-        linkable = root_view.select(Viewable)
-        linkable += root_model.select({'type' : BkModel})
-
-        if not linkable:
-            return
-
-        found = [(link, src, link.target) for src in linkable if src in cls.registry
-                 for link in cls.registry[src] if link.target in linkable or
-                 not link._requires_target]
+        linkables = root_view.select(Viewable)
+        linkables += root_model.select({'type' : BkModel})
 
         if 'holoviews' in sys.modules:
+            from holoviews import Element as HvElement
             hv_views = root_view.select(HoloViews)
             map_hve_bk = generate_panel_bokeh_map(root_model, hv_views)
-            found += [(link, src, tgt) for src in linkable if src in cls.registry
-                      for link in cls.registry[src]
-                      for tgt in map_hve_bk[link.target]]
-
+            linkables += map_hve_bk.keys()
+            if not linkables:
+                return            
+            found = [(link, src, tgt) for linkable in linkables if linkable in cls.registry
+                    for link in cls.registry[linkable]
+                    for src in ([link.source] if not isinstance(link.source, HvElement) else map_hve_bk[link.source])
+                    for tgt in ([link.target] if not isinstance(link.target, HvElement) else map_hve_bk[link.target])]
+        else:
+            if not linkables:
+                return
+            found = [(link, src, link.target) for src in linkables if src in cls.registry
+                    for link in cls.registry[src] if link.target in linkables or
+                    not link._requires_target]
+            
```

Comment:
I don't completely follow this change. Could you explain the difference?

Reply:
If you register an holoviews element as source the LinkCallBack __init__ fail because the source is an holoview element and not a Plot.
Now if we have an HvElement in link.source we find all bokeh plots(map_hve_bk[link.source]) and we create the link for each one

Reply:
Thanks.